utils/python/CIME/CreateTest.py

```diff
@@ -31,28 +33,30 @@ def __init__(self, test_names,
                  clean=False,compare=False, generate=False, namelists_only=False,
                  project=None, parallel_jobs=None):
     ###########################################################################
-        self._cime_root      = cime_util.get_cime_root()
+        self._machobj = machobj
+        self._cime_root      = CIME.utils.get_cime_root()
         self._test_names     = test_names
-        self._no_build       = no_build      if not namelists_only else True
-        self._no_run         = no_run        if not self._no_build else True
-        self._no_batch       = no_batch      if no_batch is not None else not cime_util.does_machine_have_batch()
-        self._test_root      = test_root     if test_root is not None else cime_util.get_machine_info("CESMSCRATCHROOT")
-        self._test_id        = test_id       if test_id is not None else cime_util.get_utc_timestamp()
-        self._project        = project       if project is not None else cime_util.get_machine_project()
-        self._baseline_root  = baseline_root if baseline_root is not None else cime_util.get_machine_info("CCSM_BASELINE")
```

Comment:
I understand not doing this work twice. Maybe we should handle all the default setting and argument validation here. It seems like now this class is less usable as a library because none of the arguments are defaultable anymore. For example, if the user fails to provide a test_root argument, which is legal because it defaults to None, this class will now try to use None as self._test_root which won't work.